internal/gateway/receiver/pool.go

```diff
@@ -104,11 +106,16 @@ func (p *Pool) SubmitPayload(ctx context.Context, leasePath string, payload io.R
 
 // CommitLease associated with the token (transaction commit)
 // TODO: implement timeout or context?
-func (p *Pool) CommitLease(ctx context.Context, leasePath, oldRootHash, newRootHash string, tag gw.RepositoryTag) error {
+func (p *Pool) CommitLease(ctx context.Context, leasePath, oldRootHash, newRootHash string, tag gw.RepositoryTag) (uint64, error) {
 	reply := make(chan error)
-	p.tasks <- commitTask{ctx, leasePath, oldRootHash, newRootHash, tag, reply}
+	finalRevChan := make(chan uint64, 1)
+	var finalRev uint64
```

Comment:
The channels should be closed.

I would add something like `defer close(reply)` and `defer close(finalRevChan)`.

And since we are fixing this code it would make sense to create the error channel of size 1 as well, wouldn't it?

Reply:
No, this is wrong, the worker loop, for some reason I don't understand, might send messages over the result channel during the `Quit()`.

The channels should be close, when we don't expect any message on them anymore.

Maybe we should close them in the worker loop.

Reply:
No, this is wrong, the worker loop, for some reason I don't understand, might send messages over the result channel during the Quit().

The channels should be close, when we don't expect any message on them anymore.

Maybe we should close them in the worker loop.

Reply:
I think the `task.Reply() <- err` on line 144 will actually never be read out. We should probably fix that.